content/realtime/channel-params.textile

```diff
@@ -0,0 +1,91 @@
+
+A @rewind@ parameter can express a channel position in terms of a number of messages, or a time interval.
+
+A @rewind@ value that is simply a number @n@ (eg @rewind=1@) is a request to attach to the channel at a position @n@ messages before the present position. If that attachment is successful, and one or more messages exist on the channel prior to the present position, then those messages will be delivered to the subscriber immediately after the attachment has completed, and before any subsequent messages that arise in real time.
+
+If fewer than the requested number of messages exists on the channel (including the case that there are no prior messages), then the available messages are sent; this does not constitute an error.
+
+A @rewind@ value can also be a string that is a time interval specifier. Supported specifier values express an integral number of seconds (eg @15s@), minutes (eg @5m@) or hours (eg @4h@). If that attachment is successful, and one or more messages exist on the channel in the given time interval prior to the present time, then those messages will be delivered to the subscriber immediately after the attachment has completed, and before any subsequent messages that arise in real time.
+
+If fewer than the requested number of messages exists on the channel in that interval (including the case that there are no messages), then the available messages are sent; this does not constitute an error.
+
+If the number of messages within the specified interval is greater than the attachment @limit@ (which defaults to 100), then only the most recent messages up to that limit are sent. The attachment succeeds, but truncation of the message backlog is indicated as a non-fatal error in the attachment response.
+
+It is also possible to specify a different limit via a @limit@ channel parameter. So, for example, to request up to 10 messages, in a window 5m before the present time, specify a channel parameter string of @[?rewind=5m&limit=10]@.
+
+By default, a maximum of two minutes of channel history is available when attaching. This means that a rewind time specifier of greater than two minutes will only be able to rewind by two minutes. If a channel has persistence enabled, then it is possible to rewind back in time by up to the persistence TTL on the channel. 
+
+The channel position expressed by a @rewind@ parameter has an effect only on an initial channel attachment. Any subsequent reattachment of the same channel on the same connection, in order to resume the connection, will attempt to resume with continuity from the point at which the connection dropped.
+
+Any @rewind@ parameter value that cannot be parsed either as a number or a time specifier represents an error, and any attachment request will fail with an error.
+
+h2(#examples). Examples
+
+h3(#examples-ably). Using channel parameters with the Ably library
+
+To subscribe to a channel, getting the most recent message if available:
+
+```
```

Comment:
Given we support a) interactive code samples, b) injection of keys, c) multiple languages for code samples, may I ask that we at least have one set of code samples for each transport before we hand over to devrel team?

For example, see Javascript example below from https://github.com/ably/docs/blob/master/content/realtime/channels.textile:

```[nodejs](code-editor:realtime/channel-publish)
var Ably = require(‘ably’);
var realtime = new Ably.Realtime(‘{{API_KEY}}’);
var channel = realtime.channels.get(‘{{RANDOM_CHANNEL_NAME}}’);
channel.subscribe(function(message) {
console.log("Received: " message.data);
});
channel.publish(“example”, “message data”);
```

Note that `code-editor:realtime/channel-publish` then links to https://github.com/ably/docs/blob/master/content/code/realtime/channel-publish.code, which is a full working version of the code that runs in JSBin.

Note the workflow may be easier for now to just create a JSBin directly at https://jsbin.ably.io, and include a link for Devrel to pick up to that JSBin.  For example, if you go to https://jsbin.ably.io/erobeq/1/edit which is any random JSBin which automatically injects an API key that works, you can just append `.textile` to the URL https://jsbin.ably.io/erobeq/1.textile to get the code block that goes into the docs repo.